### DISCUSSION TITLE
User-Agent String Reduction

### DISCUSSION THREAD

**Discussion by DJPVMA** [privacysandbox/privacy-sandbox-dev-support#32] [Mitigating workarounds]
I have a site at corriv.com and I have run into this issue. The avatars and cover photos are completely missing from the profile pages because of this. I really don't understand the migration hints because I am not a developer and I did not write much of this code. But, it has effected a lot of my files in this script and I have not heard back from the original developer. I do code a bit, but I have no idea what the instructions and hints are telling me. Can anyone please explain this to me so I can fix this issue on my own?

Error Message:
Audit usage of navigator.userAgent, navigator.appVersion, and navigator.platform
A page or script is accessing at least one of navigator.userAgent, navigator.appVersion, and navigator.platform. Starting in Chrome 101, the amount of information available in the User Agent string will be reduced.
To fix this issue, replace the usage of navigator.userAgent, navigator.appVersion, and navigator.platform with feature detection, progressive enhancement, or migrate to navigator.userAgentData.
Note that for performance reasons, only the first access to one of the properties is shown.

If this has been answered, I apologize, and would you please post a link to the correct thread? Thank you in advance for anything you can do to help.

**Comment by rowan-m** [comment]
Hey there, @sheriffpuma, the issue you're seeing in DevTools is an informational message about how we're [reducing the information available in the user-agent string](https://developer.chrome.com/docs/privacy-sandbox/user-agent/). These changes are being made in a way that maintains the existing format of the string, so ideally minimises where those changes might break things.

The challenge is that these features are used in a lot of places, but we can't tell if that actually would break anything - so the compromise here is that we show the "info" level message on any access.

You will need to do a little testing to determine if the user-agent reduction is actually causing the issue. You can try enabling the flag `chrome://flags/#full-user-agent` and restarting your browser to enable the full user-agent string. If the issue still occurs, it's likely there is some other cause that will need to be investigated separately. If that fixes the issue, then in DevTools you can expand the "Affected resources" section in the message to see the files that are accessing these features.
